.github/workflows/main.yaml

```diff
@@ -47,18 +52,15 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5.6.0
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: 'poetry'
-
       #----------------------------------------------
       #    install your root project, if required 
       #----------------------------------------------      
-      - name: Install library
+      - name: Install dependencies
         run: |
-          poetry install --no-interaction
+          uv sync 
+
+      - name: Verify uv.lock is up-to-date
```

Comment:
Why did you add a check of the dependencies for every change on PRs? Has the weekly check with `check-dependencies.yaml` not been sufficient? I cannot remember a problem with the weekly checks.